document-portal/meson.build

```diff
@@ -0,0 +1,63 @@
+# Note: this can be flattened into the main meson.build file but the
+# code includes "document-portal/foo" and it's too hard to get meson and
+# autotools to agree on this
```

Comment:
The Meson equivalent of `-I$(top_srcdir) -I$(top_builddir)` seems to be to have this in the top-level `meson.build`:

```
common_include_directories = include_directories('.')
```

and then use `include_directories : common_include_directories` in libraries and executables built by subdirectories.